rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/common/RMRestInterface.java

```diff
@@ -1021,4 +1041,39 @@ void removeNodeToken(@HeaderParam("sessionid") String sessionId, @HeaderParam("n
     void setNodeTokens(@HeaderParam("sessionid") String sessionId, @HeaderParam("nodeurl") String nodeUrl,
             @QueryParam("tokens") List<String> tokens) throws NotConnectedException, RestException;
 
+    /**
+     * Get the set of all tags present in all nodes
+     * @param sessionId current session
+     * @return a set of all nodes tags
+     */
+    @GET
+    @Path("node/tags")
+    @Produces("application/json")
+    Set<String> getNodeTags(@HeaderParam("sessionid") String sessionId) throws NotConnectedException, RestException;
+
+    /**
+     * Get the tags of a specific node
+     * @param sessionId current session
+     * @param url the url of the requested node
+     * @return a set of tags for the specified node
+     */
+    @GET
+    @Path("node/tags/search")
+    @Produces("application/json")
+    Set<String> getNodeTags(@HeaderParam("sessionid") String sessionId, @QueryParam("nodeurl") String url)
+            throws NotConnectedException, RestException;
+
+    /**
+     * Search the nodes with specific tag.
+     * @param sessionId current session
+     * @param tags the tag which the return nodes should contains, when it's not specified, all the known nodes urls are returned
```

Comment:
-> a list of tags which the nodes should contain. When not specified or an empty list, all the nodes known urls ...